lib/services/auth.dart

```diff
@@ -32,6 +33,52 @@ class AuthService {
     }
   }
 
+  Future<bool> updateAccount(String email, String firstName, String lastName, String phone) async {
+    Map<String, String> requestBody = new Map<String, String>();
+
+    final storage = new FlutterSecureStorage();
+    String token = await storage.read(key: 'auth_token');
+
+    requestBody['email'] = email;
+    requestBody['first_name'] = firstName;
+    requestBody['last_name'] = lastName;
+    requestBody['phone'] = phone;
+
+    if (token != null) {
+      final response = await http.patch(
+          'https://claytoncornett.tk/api/user/',
+          headers: {HttpHeaders.authorizationHeader: "Token $token"}
+      );
+      print(response.statusCode);
+      final responseBody = jsonDecode(response.body);
+      print(requestBody);
+      if (response.statusCode == 200) {
+        //await AccountRepository.getInstance().getAccount();
+//        await AccountRepository.getInstance().invalidateCurrentAccount();
+        storage.write(key: 'auth_token', value: token);
+        return true;
+      } else {
+        var errors = [];
+        var errorKeys = [
+          'email',
+          'first_name',
+          'last_name',
+          'phone',
+          'non_field_errors'
+        ];
+
+        responseBody.forEach((key, value) {
+          if (errorKeys.contains(key)) {
+            errors += responseBody[key];
+          }
+        });
+
+        return false;
```

Comment:
you may want to return more details about errors if it fails. Otherwise it'll just be a generic error and the user won't know if their input was invalid or if there is something wrong with the server, you know? Just an idea, but probably not pertinent to have done before the presentation